fedimint-client/src/lib.rs

```diff
@@ -3206,6 +3206,54 @@ impl ClientBuilder {
     }
 }
 
+/// Represents money we may be receiving, but haven't yet
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Deserialize, Serialize)]
+pub struct InFlightAmounts {
+    /// Receives that are not yet reflected in our main balance
+    pub incoming: Amount,
+    /// Sends that may be refunded
+    pub outgoing: Amount,
+}
+
+impl Default for InFlightAmounts {
+    fn default() -> Self {
+        InFlightAmounts {
+            incoming: Amount::ZERO,
+            outgoing: Amount::ZERO,
+        }
+    }
+}
+
+impl InFlightAmounts {
+    pub fn incoming(incoming: Amount) -> Self {
+        InFlightAmounts {
+            incoming,
+            outgoing: Amount::ZERO,
+        }
+    }
+
+    pub fn outgoing(outgoing: Amount) -> Self {
+        InFlightAmounts {
+            incoming: Amount::ZERO,
+            outgoing,
+        }
+    }
+}
+
+impl AddAssign for InFlightAmounts {
+    fn add_assign(&mut self, rhs: Self) {
+        self.incoming += rhs.incoming;
+        self.outgoing += rhs.outgoing;
+    }
+}
+
+impl SubAssign for InFlightAmounts {
+    fn sub_assign(&mut self, rhs: Self) {
+        self.incoming -= rhs.incoming;
```

Comment:
underflow waiting to happening

Reply:
I would rather store BTreeMap<OperationId, InFlightAmounts>, feels less error prone